pkg/metrics/lbc/collector.go

```diff
@@ -37,3 +85,99 @@ func (c *collector) ObservePodReadinessGateReady(namespace string, tgbName strin
 		labelName:      tgbName,
 	}).Observe(duration.Seconds())
 }
+
+func (c *collector) UpdateManagedK8sResourceMetrics(ctx context.Context) error {
+	listOpts := &client.ListOptions{
+		Namespace: "",
+	}
+	ingressCount, serviceCount, tgbCount := 0, 0, 0
+	// Fetch ingress count
+	ingressList := &networkingv1.IngressList{}
+	err := c.runtimeClient.List(ctx, ingressList, listOpts)
+	if err != nil {
+		return err
+	}
+	for _, ingress := range ingressList.Items {
+		for _, finalizer := range ingress.Finalizers {
+			if strings.Contains(finalizer, c.finalizerKeyWord) {
+				ingressCount++
+				break
+			}
+		}
+	}
+	c.instruments.managedIngressCount.Set(float64(ingressCount))
+
+	// Fetch service count
+	serviceList := &corev1.ServiceList{}
+	err = c.runtimeClient.List(ctx, serviceList, listOpts)
+	if err != nil {
+		return err
+	}
+	for _, service := range serviceList.Items {
+		hasMatchingFinalizer := false
+		for _, finalizer := range service.Finalizers {
+			if strings.Contains(finalizer, c.finalizerKeyWord) {
+				hasMatchingFinalizer = true
+				break
+			}
+		}
+
+		if hasMatchingFinalizer && service.Spec.LoadBalancerClass != nil && strings.Contains(*service.Spec.LoadBalancerClass, networkLoadBalancerStr) {
+			serviceCount++
+		}
+	}
+	c.instruments.managedServiceCount.Set(float64(serviceCount))
+
+	// Fetch TargetGroupBinding count
+	tgbList := &elbv2api.TargetGroupBindingList{}
+	err = c.runtimeClient.List(ctx, tgbList, listOpts)
+	if err != nil {
+		return err
+	}
+	for _, tgb := range tgbList.Items {
+		for _, finalizer := range tgb.Finalizers {
+			if strings.Contains(finalizer, c.finalizerKeyWord) {
+				tgbCount++
+				break
+			}
+		}
+	}
+	c.instruments.managedTGBCount.Set(float64(tgbCount))
+
+	return nil
+}
+
+func (c *collector) UpdateManagedALBMetrics(ctx context.Context) error {
+	count, err := c.getManagedAWSResourceMetrics(ctx, resourceTypeALB)
+	if err != nil {
+		return err
+	}
+	c.instruments.managedALBCount.Set(float64(count))
+	return nil
+}
+
+func (c *collector) UpdateManagedNLBMetrics(ctx context.Context) error {
+	count, err := c.getManagedAWSResourceMetrics(ctx, resourceTypeNLB)
+	if err != nil {
+		return err
+	}
+	c.instruments.managedNLBCount.Set(float64(count))
+	return nil
+}
+
+func (c *collector) getManagedAWSResourceMetrics(ctx context.Context, resourceType string) (count int, err error) {
+	req := &rgtsdk.GetResourcesInput{
+		ResourceTypeFilters: []string{resourceType},
+		TagFilters: []rgttypes.TagFilter{
+			{
+				Key:    awssdk.String(c.clusterTagKey),
+				Values: []string{c.clusterTagVal},
+			},
+		},
+	}
+	resources, err := c.rgt.GetResourcesAsList(ctx, req)
```

Comment:
we should not call AWS APIs to get those counter metrics. This can cause significant performance impact when there are large amount of LBs. 
It shall be technical possible to get the number of LBs managed by the controller without using k8s apis.

Reply:
Thanks! I used RGT API call, so it's just 1 api call every 2min. But I do agree it's better to update the metrics per CRUD event. let me double check and get back.